### PR TITLE
Merging with foldamers

### PR DESCRIPTION
**Pull request by cwalker7**
## Description
The big merge of CG_openmm/foldamers.

Structural analysis code (such as clustering, angle distributions) went to a new repo called analyze_foldamers, while everything else from foldamers went here. (See https://github.com/shirtsgroup/foldamers/issues/44) From now on, the foldamers repo will act as an archive, though it seems that the commit history of everything transferred is preserved. This was all done using git filter-branch --subdirectory-filter commands.

I deleted a lot of the structure files in examples - in particular, varying_model_parameters/scan_torsions had ~3700 files.

We will also need to remake the structure library so it is empty for now. All of the original structures are preserved in foldamers. 

## Todos
  - [x] Check if the updated imports are right
  - [x] Clean up the rest of the structure files (these are all saved in foldamers)
  - [x] Sort out which functions of parameters.secondary_structure should stay here

## Status
- [x] Ready to go

### CODE REVIEW

File: cg_openmm/build/cg_build.py [hunk 1]
```diff
@@ -495,8 +517,8 @@ def test_force(cgmodel, force, force_type=None):
         for nonbonded_interaction in cgmodel.nonbonded_interaction_list:
             particle_1_positions = cgmodel.positions[nonbonded_interaction[0]]
             particle_2_positions = cgmodel.positions[nonbonded_interaction[1]]
-            sigma = cgmodel.get_particle_sigma(nonbonded_interaction[0])
-            epsilon = cgmodel.get_particle_epsilon(nonbonded_interaction[0])
```
Comment (mrshirts) [inline cg_openmm/build/cg_build.py:499]: Is this reverting some of the changes recently added?  get_particle_sigma is the new version of the function, not get_sigma

File: cg_openmm/utilities/iotools.py [hunk 2]
```diff
@@ -119,27 +119,54 @@ def write_pdbfile_without_topology(CGModel, filename, energy=None):
...
-            particle_type_name = particle_type_name[0:3]
-        monomer_name = CGModel.get_particle_monomer_type(particle)["monomer_name"]
-        if len(monomer_name) > 3:
-            monomer_name = monomer_name[0:3]
-        monomer_index = CGModel.get_particle_monomer(particle)+1
-        if old_monomer_index < monomer_index:
-            old_monomer_index = monomer_index
-            element_index = 1
-        pdb_object.write(
-            f"ATOM{bead_index+1:>7d} {particle_type_name:>3s}{element_index} {monomer_name:>3s} A{monomer_index:>4}    "
```
Comment (mrshirts) [inline cg_openmm/utilities/iotools.py:136]: This looks like code being overwritten as well with the old version

File: examples/evaluating_folding/native_contacts/calculate_Q.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
+###
```
Comment (mrshirts) [inline examples/evaluating_folding/native_contacts/calculate_Q.py:1]: this one might not currently work.

File: examples/replica_exchange/run_replica_exchange.py [hunk 4]
```diff
@@ -1,9 +1,9 @@
-#!~/anaconda3/bin/python
+#!/Users/mrshirts/anaconda3/bin/python
```
Comment (mrshirts) [inline examples/replica_exchange/run_replica_exchange.py:1]: Definitely shouldn't have my path here!  Check what the most recent version is.

**Comment by mrshirts** [comment]
Also, can probably not move over a lot of the ensembles of pdbs - those will be stored for now in the old foldamers repository, we don't need to move them over.

**Comment by cwalker7** [comment]
Hmm yes it seems like there was some issue pulling from the most recent version of cg_openmm. I will try that again.

**Comment by cwalker7** [comment]
The files in question are reverted to the general_heteropolymer commit - should be fixed now. It seems to have been only 4 files affected: examples/simulation.py, examples/run_replica_exchange.py, build/cg_build.py, and utilities/iotools.py.

**Comment by cwalker7** [comment]
I got rid of a bunch more pdb files as well.

**Pull request merged**